packages/wallet-sdk/src/sign/util.ts

```diff
@@ -28,7 +28,8 @@ export async function fetchSignerType(params: {
     event: 'selectSignerType',
     data: params.preference,
   };
-  const { data } = await communicator.postMessage(request);
+  await communicator.postMessage(request);
+  const { data } = await communicator.onMessage(({ requestId }) => requestId === request.id);
```

Comment:
Is there a possible, but very unlikely, race condition between these two where the response may come back faster than the communicator starts to await?

A way around this could be to start listening first.

This is existing behavior, and don't think it's a big deal either way.

Reply:
will address.